pkg/config/global/configurer.go

```diff
@@ -0,0 +1,87 @@
+package global
+
+import (
+	"fmt"
+	"os"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	configMapName   = "kapp-controller-config"
+	caCertsKey      = "caCerts"
+	systemCertsFile = "/etc/ssl/certs/ca-certificates.crt"
+)
+
+type GlobalConfigurer struct {
+	namespace string
+	client    kubernetes.Interface
+}
+
+func NewGlobalConfigurer() (*GlobalConfigurer, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	kubeConf := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+	namespace, _, err := kubeConf.Namespace()
+	if err != nil {
+		return nil, fmt.Errorf("Getting namespace: %s", err)
+	}
+
+	restConfig := config.GetConfigOrDie()
+	coreClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Buidling client: %s", err)
+	}
+
+	return &GlobalConfigurer{
+		namespace: namespace,
+		client:    coreClient,
+	}, nil
+}
+
+func (gc *GlobalConfigurer) Configure() error {
+	configMap, err := gc.configMap()
+	if err != nil {
+		return fmt.Errorf("Fetching config map: %s", err)
+	}
+
+	if configMap == nil {
+		return nil
+	}
+
+	err = gc.addTrustedCerts(configMap.Data[caCertsKey])
+	if err != nil {
+		return fmt.Errorf("Adding trusted certs: %s", err)
+	}
+
+	return nil
+}
+
+func (gc *GlobalConfigurer) addTrustedCerts(certChain string) error {
+	if certChain == "" {
+		return nil
+	}
+
+	file, err := os.OpenFile(systemCertsFile, os.O_CREATE|os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	if err != nil {
+		return fmt.Errorf("Opening certs file: %s", err)
+	}
+	defer file.Close()
```

Comment:
i think we care about error returned from file.Close in this case since it may indicate that contents arent successfully flushed.